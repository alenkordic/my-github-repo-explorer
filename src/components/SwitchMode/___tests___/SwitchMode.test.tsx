import React from "react";
import SwitchMode from "../SwitchMode";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import { ThemeContext } from "./../../../contexts/theme.context";
import renderer from "react-test-renderer";


function switchTheme(darkMode: boolean) {
    const value = {
        toggleDarkMode: jest.fn(),
        darkMode: darkMode
      };
    return (
      <ThemeContext.Provider value={value}>
        <BrowserRouter>
          <SwitchMode />
        </BrowserRouter>
      </ThemeContext.Provider>
    );
  }
  
test("Dark is mode active", () => {
    render(switchTheme(true));
    expect(screen.getByText("Dark mode")).toBeTruthy();
  });

test("Light is mode active", () => {
    render(switchTheme(false));
    expect(screen.getByText("Light mode")).toBeTruthy();
  });

  // Snapshots
it("Switch Dark is mode render correctly", () => {
    const darkSwitch = switchTheme(true);
    const tree = renderer.create(darkSwitch).toJSON();
    expect(tree).toMatchSnapshot();
  });
it("Switch Light is mode render correctly", () => {
    const lightSwitch = switchTheme(false);
    const tree = renderer.create(lightSwitch).toJSON();
    expect(tree).toMatchSnapshot();
  });