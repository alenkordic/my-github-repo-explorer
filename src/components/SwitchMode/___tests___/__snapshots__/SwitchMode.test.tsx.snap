// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Dark is mode render correctly 1`] = `
<div
  className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
>
  <label
    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom css-77as7c-MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root MuiSwitch-sizeMedium css-lu9khh-MuiSwitch-root"
    >
      <span
        className="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={true}
          className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
      </span>
      <span
        className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1j8847n-MuiTypography-root"
    >
      Dark mode  
    </span>
  </label>
</div>
`;

exports[`Switch Light is mode render correctly 1`] = `
<div
  className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
>
  <label
    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom css-77as7c-MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root MuiSwitch-sizeMedium css-lu9khh-MuiSwitch-root"
    >
      <span
        className="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1j8847n-MuiTypography-root"
    >
      Light mode  
    </span>
  </label>
</div>
`;
